logging/tests/unit/handlers/test_app_engine.py

```diff
@@ -24,34 +25,47 @@ def _get_target_class(self):
         return AppEngineHandler
 
     def _make_one(self, *args, **kw):
-        import tempfile
+        return self._get_target_class()(*args, **kw)
 
+    def test_ctor(self):
+        import os
         from google.cloud._testing import _Monkey
-        from google.cloud.logging.handlers import app_engine as _MUT
+        from google.cloud.logging.handlers.app_engine import _GAE_PROJECT_ENV, _GAE_SERVICE_ENV, _GAE_VERSION_ENV
 
-        tmpdir = tempfile.mktemp()
-        with _Monkey(_MUT, _LOG_PATH_TEMPLATE=tmpdir):
-            return self._get_target_class()(*args, **kw)
+        client = _Client(self.PROJECT)
+        with _Monkey(os, environ={_GAE_PROJECT_ENV: 'test_project',
```

Comment:
Can you instead use:

```python
with mock.patch('os.environ', new=your_environ):
    ...
```

(We want to completely drop `_Monkey` at some point.)